craft_application/application.py

```diff
@@ -161,6 +161,20 @@ def __init__(
         # The kind of sessions that the fetch-service service should create
         self._fetch_service_policy = "strict"
 
+        # Load the plugins after initialization but before run
+        self._load_plugins()
+
+    @final
+    def _load_plugins(self) -> None:
+        """Load application plugins."""
+        # https://packaging.python.org/en/latest/specifications/entry-points/#data-model
+        for plugin_entry_point in metadata.entry_points(
+            group="craft_application_plugins.application"
+        ):
+            craft_cli.emit.progress(f"Loading app plugin {plugin_entry_point.name}")
```

Comment:
I think it's a `debug` or maybe even `trace`, not `progress` as we don't want to show that to the user.
```suggestion
            craft_cli.emit.debug(f"Loading app plugin {plugin_entry_point.name}")
```